server/main.py

```
@@ -9,15 +10,70 @@
+    except Exception as e:
+        logger.error(f"Protobuf communication error: {e}")
+
+
+async def broadcast(data):
+    """Sends binary data to all active connections."""
+    for send in list(CONNECTIONS):
+        try:
+            res = send(data)
+            if inspect.isawaitable(res):
+                await res
+        except Exception:
```

Ruff (0.15.5)

[warning] Do not catch blind exception: `Exception`

(BLE001)